src/Share/Survey/SurveyHelper.cs

```diff
@@ -0,0 +1,353 @@
+        {
+            helper.Modules[moduleName].LastActiveDate = CurrentDate;
+            helper.Modules[moduleName].ActiveDays += 1;
+        }
+
+        private bool ShouldModulePromptExpire(SurveyHelper helper, string moduleName, int majorVersion)
+            => helper.Modules[moduleName].MajorVersion == majorVersion
+            && (Modules[moduleName].ActiveDays == SurveyTriggerCount + 1 && helper.LastPromptDate == Convert.ToDateTime(Modules[moduleName].LastActiveDate) && helper.Today > helper.LastPromptDate.AddDays(DelayForSecondPrompt)
+                || Modules[moduleName].ActiveDays == SurveyTriggerCount + 2 && helper.LastPromptDate == Convert.ToDateTime(Modules[moduleName].LastActiveDate) && helper.Today > LastPromptDate.AddDays(DelayForThirdPrompt));
+
+        private void ModulePromptExpire(SurveyHelper helper, string moduleName, int majorVersion)
+        {
+            helper.Modules[moduleName].ActiveDays = 0;
+        }
+
+        private void MergeScheduleInfo(ScheduleInfo externalScheduleInfo)
+        {
+            DateTime externalLastPromptDate = Convert.ToDateTime(externalScheduleInfo?.LastPromptDate);
+            IDictionary<string, ModuleInfo> externalMap = new Dictionary<string, ModuleInfo>();
+            foreach(ModuleInfo info in externalScheduleInfo?.Modules)
+            {
+                externalMap[info.Name] = info;
+            }
+
+            HashSet<string> moduleNames = new HashSet<string>(Modules.Keys);
+            moduleNames.UnionWith(new HashSet<string>(externalMap.Keys));
+
+            foreach (string name in moduleNames)
+            {
+                ModuleInfo item = null;
+                if (Modules.ContainsKey(name) && (!externalMap.ContainsKey(name) || Convert.ToDateTime(Modules[name].LastActiveDate) > Convert.ToDateTime(externalMap[name].LastActiveDate)))
+                {
+                    item = new ModuleInfo(Modules[name]);
+                }
+                else
+                {
+                    if (externalLastPromptDate != DateTime.MinValue && Convert.ToDateTime(externalMap[name].LastActiveDate) == externalLastPromptDate)
+                    {
+                        LastPromptDate = externalLastPromptDate;
+                    }
+                    item = new ModuleInfo(externalMap[name]);
+                    Modules[name] = item;
+                }
+            }
+        }
+
+        private ScheduleInfo GetModules()
+        { 
+            return new ScheduleInfo() { LastPromptDate = LastPromptDate.ToString("yyyy-MM-dd"), Modules = Modules.Values.ToList() };
+        }
+
+        private bool ReadFromStream()
+        {
+            StreamReader sr = null;
+            try
+            {
+                if (File.Exists(SurveySchedulePath))
+                {
+                    sr = new StreamReader(new FileStream(SurveySchedulePath, FileMode.Open, FileAccess.Read, FileShare.None));
```

Comment:
Please use this API https://docs.microsoft.com/en-us/dotnet/api/system.io.file.readalltext?view=net-5.0
You don't need to read it via Stream.